### PR TITLE
Update arrays in roslyn to use collection expressions

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This is only for array literals.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -280,3 +280,5 @@ dotnet_diagnostic.IDE2006.severity = warning
 # CA1822: Make member static
 # There is a risk of accidentally breaking an internal API that partners rely on though IVT.
 dotnet_code_quality.CA1822.api_surface = private
+
+dotnet_diagnostic.IDE0055.severity = none
```
Comment (CyrusNajmabadi) [inline .editorconfig:284]: Will file issue to reenable this once we can once the formatting analyzer is fixed up to support collection expressions.

File: eng/config/globalconfigs/Common.globalconfig [hunk 2]
```diff
@@ -11,7 +11,7 @@ dotnet_diagnostic.CA2009.severity = warning
 dotnet_diagnostic.CA2012.severity = warning
 
 # IDE0055: Fix formatting
-dotnet_diagnostic.IDE0055.severity = warning
+dotnet_diagnostic.IDE0055.severity = none
```
Comment (CyrusNajmabadi) [inline eng/config/globalconfigs/Common.globalconfig:14]: Will file issue to reenable this once we can once the formatting analyzer is fixed up to support collection expressions.

File: .editorconfig (hunk 1 above)
Comment (sharwell) [reply .editorconfig]: This should be disabled for only a select group of files (e.g. 2-3 total files in the solution), and not for the entire solution.

**Comment by CyrusNajmabadi** [comment]
Sorry, if i disable that, over 90 files get reported as having issues.  You
can see them here:  Demonstrating broken formatting. by CyrusNajmabadi ·
Pull Request #70551 · dotnet/roslyn (github.com)
<https://github.com/dotnet/roslyn/pull/70551>

This needs to be disabled as it's blocking over a 3rd of the cases from
being updated.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide ptal.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide ptal

**Comment by CyrusNajmabadi** [comment]
https://github.com/orgs/dotnet/teams/roslyn-ide @ToddGrun ptal

**Comment by CyrusNajmabadi** [comment]
https://github.com/orgs/dotnet/teams/roslyn-ide @ToddGrun ptal

File: .editorconfig [hunk 3]
```diff
@@ -149,7 +149,8 @@ dotnet_public_api_analyzer.require_api_files = true
 
 # IDE0055: Fix formatting
 # Workaround for https://github.com/dotnet/roslyn/issues/70570
-dotnet_diagnostic.IDE0055.severity = warning
+# Disabled because the current SDK warns on collection expressions
+dotnet_diagnostic.IDE0055.severity = none
```
Comment (ToddGrun) [inline .editorconfig:153]: Is it already fixed in the sdk? If not, is there a bug? If so, is updating the SDK an option?

Comment (CyrusNajmabadi) [reply .editorconfig:153]: it's fixed in the SDK.  however, we cannot update the SDK for reasons that are never quite clear to me.  so we're stuck in this state.  Being able to dogfood trumps using an old SDK though, so this has to happen.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**